common/src/dynamic_amm/ix_account_builder.rs

```diff
@@ -0,0 +1,461 @@
+use crate::{dynamic_amm::pda::*, dynamic_vault::pda::*};
+use anchor_lang::AccountDeserialize;
+use anchor_spl::associated_token::get_associated_token_address;
+use dynamic_amm::state::{Depeg, TokenMultiplier};
+use dynamic_vault::state::Vault;
+use solana_sdk::{account::Account, pubkey::Pubkey};
+use std::future::Future;
+
+pub enum CurveTypeIx {
+    ConstantProduct,
+    Stable,
+}
+
+impl From<CurveTypeIx> for dynamic_amm::state::CurveType {
+    fn from(value: CurveTypeIx) -> Self {
+        match value {
+            CurveTypeIx::ConstantProduct => dynamic_amm::state::CurveType::ConstantProduct,
+            CurveTypeIx::Stable => dynamic_amm::state::CurveType::Stable {
+                amp: 0,
+                token_multiplier: TokenMultiplier::default(),
+                depeg: Depeg::default(),
+                last_amp_updated_timestamp: 0,
+            },
+        }
+    }
+}
+
+fn get_or_derive_vault_related_keys(
```

Comment:
logic here is a bit weird, we can just hardcode some vault that has lp_mint is not PDA
SOL, USDC, USDT

Reply:
Fixed